pkg/pool-utils/contracts/controllers/BasePoolController.sol

```diff
@@ -61,27 +192,74 @@ contract BasePoolController is IBasePoolController, Ownable {
     }
 
     /**
-     * @dev Pass a call to BasePool's setSwapFeePercentage through to the underlying pool
+     * @dev Stores the proposed new manager in `_managerCandidate`. To prevent accidental transfer to an invalid
+     * address, the candidate address must call `claimOwnership` to complete the transfer.
+     *
+     * Can only be called by the current manager.
+     */
+    function transferOwnership(address newManager) external onlyManager {
+        _require(canTransferOwnership(), Errors.UNAUTHORIZED_OPERATION);
+
+        _managerCandidate = newManager;
+    }
+
+    /**
+     * @dev This must be called by the manager candidate to complete the transferwnership operation.
+     * This "claimable" mechanism prevents accidental transfer of ownership to an invalid address.
+     *
+     * To keep this function simple and focused, transferring ownership does not affect the swapFeeController.
+     * Sometimes the new owner might want to retain the "old" swap fee controller (e.g., if it was
+     * delegated to Gauntlet). Other times an owner may want to take control of fees from the previous
+     * owner. In the latter case, the new owner should call `setSwapFeeController`.
+     */
+    function claimOwnership() external {
+        address candidate = _managerCandidate;
+
+        _require(candidate == msg.sender, Errors.SENDER_NOT_ALLOWED);
+
+        emit OwnershipTransferred(_manager, candidate);
+        _manager = candidate;
+
+        // Setting the candidate to zero prevents calling this repeatedly and generating multiple redundant events,
+        // and also allows checking (perhaps by a UI) whether there is a pending transfer.
+        _managerCandidate = address(0);
+    }
+
+    /**
+     * @dev Change the address allowed to call setSwapFeePercentage.
      */
-    function setSwapFeePercentage(uint256 swapFeePercentage) external virtual override onlyOwner withBoundPool {
+    function setSwapFeeController(address newSwapFeeController) external onlyManager {
+        emit SwapFeeControllerChanged(getSwapFeeController(), newSwapFeeController);
+
+        _swapFeeController = newSwapFeeController;
+    }
+
+    /**
+     * @dev Pass a call to BasePool's setSwapFeePercentage through to the underlying pool, if allowed.
+     */
+    function setSwapFeePercentage(uint256 swapFeePercentage) external virtual override withBoundPool {
+        _require(canChangeSwapFee(), Errors.UNAUTHORIZED_OPERATION);
+        _require(getSwapFeeController() == msg.sender, Errors.SENDER_NOT_ALLOWED);
```

Comment:
Do we want to leave the option of setting the swap fee controller to `0xba1...ba1` to delegate it to governance? 

We'd need to have similar logic to check with the authorizer as in the pools in that case.

Reply:
I was thinking about that, but balked because it would need that extra logic. The idea was a project might contract with a provider themselves (vs piggy-backing on our current Gauntlet agreement, which has limited slots, etc.), and delegate to whatever actual address they use to control the fees, vs our governance placeholder address.

Open for discussion, though. I thought keeping it simple and generic would scale better. They can write a bot and delegate to that address, or still use a provider like Gauntlet - but the onus is on them to create the relationship.

Reply:
Yeah, I don't think that this would be a particularly commonly used feature.

If we exposed a `canPerform` function on the pools we could piggy back of that but seems like we'd have to have a whole separate mechanism (as there's no easy way to get the pool's factory address) which is definitely in "not worth it" territory.

Reply:
I was just making sure that we considered this first but I agree with your take.